scRNA-seq/01-filtering_scRNA-seq.Rmd

```diff
@@ -0,0 +1,420 @@
+
+
+
+```{r filepaths}
+# main data directory
+data_dir <- file.path("data", "glioblastoma")
+
+# Count matrix file location
+gene_matrix_file <- file.path(data_dir,
+                              "preprocessed",
+                              "tximport", 
+                              "count_matrix.tsv")
+
+# Metadata file location
+metadata_file <- file.path(data_dir, "preprocessed", "darmanis_metadata.tsv")
+
+# Mitochondrial genes
+mito_genes_file <- file.path(data_dir, "hs_mitochondrial_genes.tsv")
+
+# Output file 
+filtered_count_file <- file.path("data", "glioblastoma", "filtered_count_matrix.tsv")
+```
+
+Let's import the gene matrix and metadata. 
+
+```{r read_data}
+# Read in the data
+sc_data <- readr::read_tsv(gene_matrix_file, progress = FALSE)
+
+# Read in the metadata
+sc_metadata <- readr::read_tsv(metadata_file, progress = FALSE)
+```
+
+Now that we have loaded our data, let's take a look at the general format
+Use a command we have used previously to look at the first few rows of `sc_data`.
+
+```{r peek_data, live = TRUE}
+# Put a command here that would allow you to see the first few rows of `sc_data`
+head(sc_data)
+```
+
+You'll notice the first column contains the gene information and the rest are samples. 
+
+Do the sample ids match what we have in the metadata? 
+For now, we will not worry about order, but just check that we have metadata for every sample.
+
+```{r sample_match, live = TRUE}
+samples <- colnames(sc_data)[-1]
+all(samples %in% sc_metadata$geo_accession)
+```
+
+
+For the analyses in this notebook, the gene column (which is a `character` type), will get in the way, so we will turn this column into the rownames instead, and convert the rest to a matrix.
+Recall that we can do this because all the remaining data has the same type.
+While not strictly necessary, having the data in the form of a matrix will make some calculations more efficient.
+
+```{r make_matrix}
+# Set the column as the gene names
+sc_data <- tibble::column_to_rownames(sc_data, "gene")
+sc_data <- as.matrix(sc_data)
+```
+
+Let's look at the mean expression of the genes in this dataset. 
+We will use `apply()` in order to calculate things across our data frame. 
+The second argument in `apply()` specifies whether we are calculating by rows or columns. 
+(1 = rows, 2 = columns).
+
+In the code chunk below, use `apply()` with the correct arguments to calculate the gene means. 
+
+```{r means, live = TRUE}
+# Let's calculate the gene means (by row)
+gene_means <- apply(sc_data, 1, mean)
+```
+
+This works just fine, but you may have noticed it is a bit slow.
+For a few common summary functions like means and sums, R has much more efficient functions to calculate across rows or columns. 
+In this case, we can use `rowMeans()` to do the same calculation, much more quickly.
+
+
+```{r rowmeans}
+# use rowMeans() to calculate gene means
+gene_means <- rowMeans(sc_data)
+```
+
```

Comment:
```suggestion
```
I don't like the extra spacings we have in these notebooks and not the others. It really doesn't matter, I'm being nit picky, but I needed to say it.